lib/preferences.php

```diff
@@ -0,0 +1,74 @@
+<?php
+/**
+ * Preference handling.
+ *
+ * @package gutenberg
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	die( 'Silence is golden.' );
+}
+
+/***
+ * User preference management.
+ */
+class Gutenberg_User_Preferences {
+	const VALID_PREFERENCES = array(
+		'block_usage',
+		'layout_config',
+	);
+
+	/**
+	 * Validates that a preference name is valid, as part of a REST API request.
+	 *
+	 * @param  string           $preference_name Preference name.
+	 * @return bool             If the preference name is valid preference.
+	 */
+	public static function is_valid_preference_name( $preference_name ) {
+		return in_array( $preference_name, self::VALID_PREFERENCES );
```

Comment:
We can add a filter here to allow plugins to register their own user preferences.  Not something that needs to be done for this PR.